roles/docker/tasks/main.yml

```diff
@@ -10,7 +10,12 @@
 - name: Downgrade docker if necessary
   command: "{{ ansible_pkg_mgr }} downgrade -y docker-{{ docker_version }}"
   register: docker_downgrade_result
-  when: not docker_version_result | skipped and docker_version_result | default('0.0', True) | version_compare(docker_version, 'gt')
+  when: not docker_version_result | skipped and docker_version_result.stdout | default('0.0', True) | version_compare(docker_version, 'gt')
+
+- name: Upgrade docker if necessary
+  command: "{{ ansible_pkg_mgr }} upgrade -y docker-{{ docker_version }}"
+  register: docker_upgrade_result
+  when: docker_downgrade_result | skipped and docker_version_result.stdout | default('0.0', True) | version_compare(docker_version, 'lt')
```

Comment:
Is this needed? wouldn't the yum/dnf install step below handle the upgrade step?